### PR TITLE
Make request headers available to console templates.

### PR DESCRIPTION
**Pull request by pstibrany**
This patch makes headers (first value for each header) from HTTP request available to the console templates.

### CODE REVIEW

File: web/web.go [hunk 1]
```diff
@@ -282,14 +282,20 @@ func (h *Handler) consoles(w http.ResponseWriter, r *http.Request) {
...
 	data := struct {
 		RawParams url.Values
 		Params    map[string]string
 		Path      string
+		Headers   map[string]string
 	}{
 		RawParams: rawParams,
 		Params:    params,
 		Path:      strings.TrimLeft(name, "/"),
+		Headers:   headers,
```
Comment (brian-brazil) [inline web/web.go]: Similarly to Params, we should make RawHeaders available too

**Comment by brian-brazil** [comment]
May I ask why you want this?

**Comment by pstibrany** [comment]
To be able to generate some other links in top menu based on Host: header.

File: web/web.go (hunk 1 above)
Comment (pstibrany) [inline web/web.go]: I had it there, but removed it :) I can add it back. I guess documentation is also missing.

**Comment by brian-brazil** [comment]
We should be exposing the -web.external-url for that, rather than full headers as you likely need the path too. Look at how `pathPrefix` is done.

Couldn't you use absolute paths?

**Comment by pstibrany** [comment]
Sorry for not being specific enough. We have several Prometheus instances running in different DCs. Our idea is to prepare console templates, and put some additional links on the top-menu, pointing to other DC-specific pages, which are outside of Prometheus. Using Host: header for that looked like a simplest option, and exposing Headers to console templates sounded like something that other people may find useful.

**Comment by brian-brazil** [comment]
I don't see a fundamental problem with exposing the headers, though it does seem a tad odd.

@grobie Do you have an opinion here based on the external-url stuff?

**Comment by pstibrany** [comment]
Btw, we *do* use -web.external-url, but we set it to value internal value (-web.external-url http://prom:9090/prometheus). We then access Prometheus through proxy, or through ssh tunnel (and then via http://localhost:<ssh tunnel>/prometheus). It seems that Prometheus doesn't really use host part of external URL, so both work for us.

**Comment by grobie** [comment]
Interesting use case. I don't really have an opinion here, mostly as we don't use consoles (I started to use Grafana's scripting features in order to programmatically build dashboards). I guess it's fine to expose headers, but it increases the interface of templates quite a bit.

**Comment by pstibrany** [comment]
To be honest, we ended up not using this. Closing this PR.

**Pull request closed without merging**